addons/sale_loyalty/tests/test_loyalty.py

```diff
@@ -430,3 +431,46 @@ def test_gift_card_program_without_product(self):
         self._claim_reward(order, giftcard_program)
 
         self.assertEqual(giftcard_program.coupon_count, 0)
+
+    def test_ewallet_user_restriction(self):
+        product_A = self.env['product.product'].create({
+            'name': 'Product A',
+            'list_price': 150,
+            'taxes_id': [],
+        })
+
+        partner_b = self.env['res.partner'].create({'name': 'Jean Paul'})
+
+        ewallet_program = self.env['loyalty.program'].create([{
+            'name': 'E-wallet Card Program',
+            'program_type': 'ewallet',
+            'trigger': 'auto',
+            'applies_on': 'current',
+            'rule_ids': [Command.create({
+                'reward_point_amount': 1,
+                'reward_point_mode': 'unit',
+            })],
+            'reward_ids': [Command.create({
+                'reward_type': 'discount',
+                'discount': 1,
+                'discount_mode': 'per_point',
+                'discount_applicability': 'order',
+            })],
+        }])
+
+        self.env['loyalty.generate.wizard'].with_context(active_id=ewallet_program.id).create({
```

Comment:
I feel like it would make it clearer if you assign the coupon to another partner (even if not necessary).